### ISSUE TITLE
"Block not downloaded" for cloning an archive still being creating

### ISSUE THREAD

**Issue by clkao** [datproject/dat#798]
1. Share a folder with large number of files:

```
$ du -sh 
645M	.

$ dat share .
dat v13.6.0
Created new dat in /Users/clkao/src/st/stconfig/git/jirs/.dat
dat://e05ca4d7034e00f3769360010f153dc85d56ebe29e45c1033759eda9ea7653a5
Sharing dat: 39636 files (216 MB)

0 connections | Download 0 B/s Upload 0 B/s

Creating metadata for 77034 files (177 KB/s)
[===================-----------------------] 46%
ADD: 200910/  臺中地方法院刑 8,簡690,20091006,1.json (11 KB)

```

wait until the initial file count is done.

2. In a second terminal, clone the archive:
```
dat clone dat://e05ca4d7034e00f3769360010f153dc85d56ebe29e45c1033759eda9ea7653a5 jirs
dat v13.6.0
Created new dat in /private/tmp/jirs/.dat
Cloning: (empty archive)

1 connection | Download 1.2 MB/s Upload 0 B/s

Downloading updates...
[=-----------------------------------------] 4.31%

Ctrl+C to Exit
events.js:160
      throw er; // Unhandled 'error' event
      ^

Error: Block not downloaded
    at Feed.get (/Users/clkao/.nvm/versions/node/v6.10.3/lib/node_modules/dat/node_modules/hypercore/index.js:785:48)
    at Tree._getAndDecode (/Users/clkao/.nvm/versions/node/v6.10.3/lib/node_modules/dat/node_modules/append-tree/index.js:541:13)
    at Tree.head (/Users/clkao/.nvm/versions/node/v6.10.3/lib/node_modules/dat/node_modules/append-tree/index.js:354:47)
    at Tree.list (/Users/clkao/.nvm/versions/node/v6.10.3/lib/node_modules/dat/node_modules/append-tree/index.js:108:8)
    at /Users/clkao/.nvm/versions/node/v6.10.3/lib/node_modules/dat/node_modules/append-tree/index.js:475:16
    at onnodes (/Users/clkao/.nvm/versions/node/v6.10.3/lib/node_modules/dat/node_modules/append-tree/index.js:126:7)
    at get (/Users/clkao/.nvm/versions/node/v6.10.3/lib/node_modules/dat/node_modules/append-tree/index.js:564:10)
    at _combinedTickCallback (internal/process/next_tick.js:83:11)
    at process._tickCallback (internal/process/next_tick.js:104:9)
```

**Comment by mafintosh** [comment]
I know what is causing this. Will try and push a fix tmw

**Comment by mafintosh** [comment]
@clkao can you reproduce this?

**Comment by clkao** [comment]
@mafintosh yes this is quite consistent with a large archive still being shared.  Or do you mean that I should try any updated modules?

**Comment by mafintosh** [comment]
@clkao Okay cool, thats good. I have a hard time reproducing myself. Could you share a video of what you do to make this happen?

**Comment by mafintosh** [comment]
@clkao follow up: try this: when it fails kill the source process and run the clone again (with no other dat process) and see if the clone throws an error

**Comment by mafintosh** [comment]
If the clone fails consistently could you zip up the .dat folder and share with me?

**Comment by mafintosh** [comment]
Reinstall node_modules. There is a high chance that this is fixed in hyperdrive 9.4.1 which I just released.

**Comment by joehand** [comment]
Hopefully this is fixed. Please reopen if still a bug.
